docs/source/specification.rst

```diff
@@ -0,0 +1,20 @@
+.. _specification:
+
+====================================================
+The Reproducible Execution Environment Specification
+====================================================
+
+repo2docker scans a repository for particular :ref:`config_files`, such
+as ``requirements.txt`` or ``REQUIRE``. The collection of files and their contents
+that repo2docker uses is known as the **Reproducible Execution Environment Specification**.
+
+The goal of the REE Specification is to provide a structure that is clearly-defined and that
+can be extended to accomodate more components of a reproducible workflow.
```

Comment:
I would add some text along the lines of "automating existing community best practices", e.g. why we support `requirements.txt` and not `repo2docker-packages.fmt`.

Reply:
How about something like this:

> The goal of REES is to automate and encourage existing community best practices for reproducible computational environments. This includes installing community-standard specification files such as `requirements.txt` or `REQUIRE` using standard tools such as `pip` or `conda` or `apt`. While repo2docker automates the creation of the environment, a human should be able to look at a REES-following repository and reproduce the environment using common, clear steps without repo2docker software.

Reply:
I like the spirit and the wording of this.